File: Functions/AreaCalculator/AreaCalculator.java
//Function AreaCalculator 110pts Nico Liao
import java.util.*;
public class AreaCalculator {
    public static void main(String[] args) {
        int choice;
        while(true) {
            System.out.println("---------------");
            System.out.println("1) Triangle");
            System.out.println("2) Rectangle");
            System.out.println("3) Square");
            System.out.println("4) Circle");
            System.out.println("5) Quit");
            System.out.print("Which Shape do you want to calculate: ");
            Scanner keyboard = new Scanner(System.in);
            choice = keyboard.nextInt();
            if (choice == 1) {
                int base, height;
                System.out.print("Base: ");
                base = keyboard.nextInt();
                System.out.print("Height: ");
                height = keyboard.nextInt();
                System.out.println("The area is " + area_triangle(base, height));
                area_triangle(base, height);
            }else if (choice == 2) {
                int length, width;
                System.out.print("Length: ");
                length = keyboard.nextInt();
                System.out.print("Width: ");
                width = keyboard.nextInt();
                System.out.println("The area is " + area_rectangle(length, width));
            }else if (choice == 3) {
                int side;
                System.out.print("Side: ");
                side = keyboard.nextInt();
                System.out.println("The area is " + area_square(side));
            }else if (choice == 4) {
                double radius;
                System.out.print("Radius: ");
                radius = keyboard.nextDouble();
                System.out.println("The area is " + area_circle(radius));
            }else if (choice == 5) {
                break;
            }
        } 
        System.out.println("Bye");
    }
    public static double area_circle( double radius ) {
        double area;
        area = Math.PI*radius*radius;
        return area;
    }
    public static int area_rectangle( int length, int width ) {
        int area;
        area = length*width;
        return area;
    }
    public static int area_square( int side ) {
        int area;
        area = side*side;
        return area;
    }
    public static double area_triangle( int base, int height ) {
        int area;
        area = base*height/2;
        return area;
    }
}

File: C++/Conversion4/AreaCalculator/AreaCalculator.cpp
//functions AreaCalculator 110pts Nico Liao
#include <iostream>
#include <cmath>

using namespace std; 
int choice = 0;
void triangle(int x, int y) {
    double area = 0;
    cout << "Base: ";
    cin >> x;
    cout << endl << "Height: ";
    cin >> y;

    area = x * y;
    area = area/2;

    cout << endl << "The area is " << area << "." << endl;
}

void rect(int x, int y) {
    int area = 0;
    cout << "Length: ";
    cin >> x;
    cout << endl << "Width: ";
    cin >> y;
    area = x * y;
    cout << endl << "The area is " << area << "." << endl;
}

void square(int x) {
    int area = 0;
    cout << "Side: ";
    cin >> x;
    area = x*x;
    cout << endl << "The area is " << area << "." << endl;
}

void circle(int x) {
    double area = 0;
    double powrad = 0;

    cout << "Radius: ";
    cin >> x;

    powrad = pow(x, 2);

    area = 3.14159*powrad;
    cout << endl << "The area is " << area << "." << endl;

}
int main(){
    void (*ptriangle)(int x, int y);
    ptriangle = triangle;
    void (*prect)(int x, int y);
    prect = rect;
    void (*psquare)(int n);
    psquare = square;
    void (*pcircle)(int n);
    pcircle = circle;
    cout << "Shape Area Calculator version 0.1 (c) 2005 Mitchell Sample Output, Inc." << endl;
    cout << "\n";
    cout << "-=-=-=-=-=-=-=-=-=-=-=-=-=-=-" << endl << endl;
    do {
        cout << "1) Triangle" << endl << "2) Rectangle" << endl << "3) Square" << endl << "4) Circle" << endl << "5) Quit" << endl << "Which shape: ";
        cin >> choice;

        switch (choice) {
            case 1: 
                ptriangle(5 , 6);
                break;
            case 2: 
                prect(5, 6);
                break;
            case 3:
                psquare(5);
                break;
            case 4:
                pcircle(5);
                break;
            case 5: 
                break;
        }
        cout << "\n";
        cout << "-=-=-=-=-=-=-=-=-=-=-=-=-=-=-" << endl << endl;
    } while (choice != 5);
    cout << "Goodbye.";

}